app/app.js

```diff
@@ -260,13 +281,29 @@ export default class AppRenderer {
 
   async updateAccountExpiry() {
     const actions = this._reduxActions;
+    const accountDataState = this._accountDataState;
+
+    // Bail if something else requested an update to account data
+    // or if account data cache was updated recently.
+    if (accountDataState.isUpdating() || !accountDataState.needsUpdate()) {
+      return;
+    }
+
     try {
       const accountToken = await this._daemonRpc.getAccount();
       if (!accountToken) {
         throw new NoAccountError();
       }
-      const accountData = await this._daemonRpc.getAccountData(accountToken);
-      actions.account.updateAccountExpiry(accountData.expiry);
+
+      const accountData = await accountDataState.update(() => {
+        return this._daemonRpc.getAccountData(accountToken);
+      });
+
+      // Check if account token is still the same after receiving account data
+      const currentAccountToken = this._reduxStore.getState().account.accountToken;
```

Comment:
Not very proud of this, but basically `get_account_data` is asynchronous and we can't cancel it, it can happen that the response arrives way too late.

As I said I am thinking to use operations that we could cancel on logout. But right now that's the cheapest possible substitution IMO.